cmd/ttn-lw-cli/commands/end_device_templates.go

```diff
@@ -0,0 +1,436 @@
+			}
+
+			var input ttnpb.EndDevice
+			decodedPaths, err := inputDecoder.Decode(&input)
+			if err != nil {
+				return err
+			}
+			decodedPaths = ttnpb.FlattenPaths(decodedPaths, endDeviceFlattenPaths)
+			decodedPaths = ttnpb.ExcludeFields(decodedPaths, excludePaths...)
+			paths = append(paths, decodedPaths...)
+
+			mappingKey, _ := cmd.Flags().GetString("mapping-key")
+			res := &ttnpb.EndDeviceTemplate{
+				FieldMask: pbtypes.FieldMask{
+					Paths: paths,
+				},
+				MappingKey: mappingKey,
+			}
+			res.EndDevice.SetFields(&input, paths...)
+
+			return io.Write(os.Stdout, config.OutputFormat, res)
+		}),
+	}
+	endDeviceTemplatesMakeCommand = &cobra.Command{
+		Use:   "make [flags]",
+		Short: "Make an end device from a template",
+		Long: `Make an end device from a template
+
+This command takes an end device template from stdin.`,
+		PersistentPreRunE: preRun(),
+		RunE: asBulk(func(cmd *cobra.Command, args []string) error {
+			if inputDecoder == nil {
+				return nil
+			}
+
+			forwardDeprecatedDeviceFlags(cmd.Flags())
+
+			var input ttnpb.EndDeviceTemplate
+			_, err := inputDecoder.Decode(&input)
+			if err != nil {
+				return err
+			}
+
+			var device ttnpb.EndDevice
+			device.SetFields(&input.EndDevice, input.FieldMask.Paths...)
+			if err = util.SetFields(&device, setEndDeviceFlags); err != nil {
+				return err
+			}
+
+			return io.Write(os.Stdout, config.OutputFormat, &device)
+		}),
+	}
+	endDeviceTemplatesAssignEUIsCommand = &cobra.Command{
+		Use:     "assign-euis [join-eui] [start-dev-eui] [flags]",
+		Aliases: []string{"euis"},
+		Short:   "Assign JoinEUI and DevEUIs to end device templates",
+		Long: `Assign JoinEUI and DevEUIs to end device templates
+
+This command takes an end device template from stdin.`,
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if inputDecoder == nil {
+				return nil
+			}
+
+			forwardDeprecatedDeviceFlags(cmd.Flags())
+			joinEUIHex, _ := cmd.Flags().GetString("join-eui")
+			startDevEUIHex, _ := cmd.Flags().GetString("start-dev-eui")
+			switch len(args) {
+			case 0:
+			case 1:
+				logger.Warn("only single EUI found in arguments, not considering arguments")
+			case 2:
+				joinEUIHex = args[0]
+				startDevEUIHex = args[1]
+			default:
+				logger.Warn("multiple EUIs found in arguments, considering the first")
+				joinEUIHex = args[0]
+				startDevEUIHex = args[1]
+			}
+			if joinEUIHex == "" {
+				return errNoEndDeviceTemplateJoinEUI
+			}
+			if startDevEUIHex == "" {
+				return errNoEndDeviceTemplateStartDevEUI
+			}
+
+			var joinEUI types.EUI64
+			if err := joinEUI.UnmarshalText([]byte(joinEUIHex)); err != nil {
+				return err
+			}
+			var startDevEUI types.EUI64
+			if err := startDevEUI.UnmarshalText([]byte(startDevEUIHex)); err != nil {
+				return err
+			}
+			devEUIInt := binary.BigEndian.Uint64(startDevEUI[:])
+
+			count, _ := cmd.Flags().GetInt("count")
+			for {
```

Comment:
Ok, so if you pipe 3 templates into this command, with `--count 100` you'd output 100 templates for each input template. Is there any practical use case for that?

Reply:
Yes, you get each template times `--count`. That's a side effect.

The reason it's implemented this way is that you can use the same command for assigning EUIs to a number of templates, or to a single template.

Still, if you have EU and US templates and you want both times 100, you can do that too. Edge case, but supported, not the intention of this command.